static/js/message_list_data.js

```diff
@@ -542,3 +549,52 @@ export class MessageListData {
         return msg;
     }
 }
+
+export function get_message_list_data(excludes_muted_topics, filter = new Filter()) {
+    /* `excludes_muted_topics` is not included in the key since we can determine `excludes_muted_topics`
+       from the filter itself. So, `excludes_muted_topics` doesn't vary with the filter.
+    */
+    const data_key = JSON.stringify(filter._operators);
```

Comment:
I don't think this will be a reliable key, since I'm not sure the order will be consistent. I wonder if we can use the "narrow URL" generated by `hash_util` as a unique string for a view -- we'd need to take some care to just have a fixed sort order in which operators are emitted into those URLs (E.g. "stream" always before "topic", which I think we might already do? Not sure).

Reply:
We use the same operators to create the hash. The order in the hash is the same as in the operators list.

See `hash_util.operators_to_hash()`:


```js
        for (const elem of operators) {
            // Support legacy tuples.
            const operator = elem.operator;
            const operand = elem.operand;

            const sign = elem.negated ? "-" : "";
            hash +=
                "/" +
                sign +
                internal_url.encodeHashComponent(operator) +
                "/" +
                encode_operand(operator, operand);
        }
```